config/v1/types_apiserver.go

```diff
@@ -0,0 +1,24 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// APIServer holds cluster-wide information about api-servers.  The canonical name is `cluster`
+type APIServer struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec APIServerSpec `json:"spec"`
+	Status APIServerStatus `json:"status"`
+}
+
+type APIServerSpec struct {
```

Comment:
@sttts would you place audit in here?  It's a possibility

Reply:
This is my understanding. But then I don't understand e.g. NamedCertificates or APIServerCertInfo. We have to carefully split the config we have today.

Reply:
* I hear audit config will be (at some point, possibly post-4.0) in its own config. (@deads2k / @smarterclayton might have more info)
* `APIServerCertInfo` & `NameCertificates` are only used by the kube-apiserver, who has exposed endpoints.

Reply:
> This is my understanding. But then I don't understand e.g. NamedCertificates or APIServerCertInfo. We have to carefully split the config we have today.

Having given this more thought, I think audit and admission would be separate and cross cutting.  Audit may be dropped in the future.